### PR TITLE
Fix health

### PR DESCRIPTION
**Pull request by pcuzner**
Multiple updates including;  gateway creation changes, gateway validity (s/w version checks) and ceph module updates to provide more health information for the ceph cluster (via an 'info' command)

### CODE REVIEW

File: gwcli/ceph.py [hunk 1]
```diff
@@ -171,24 +149,92 @@ def ui_command_refresh(self):
...
+        output += "\nConfig : {}\n".format(self.conf)
+        output += "Keyring: {}\n".format(os.path.join(CephGroup.ceph_config_dir,
+                                                      self.keyring))
+
+        console_message(output, color=color[self.health_status])
+
+    @property
+    def cluster_version(self):
+
+        vers_out = os_cmd("ceph -c {} version".format(self.conf))
```
Comment (dillaman) [inline gwcli/ceph.py]: Why not just use the Python API's version method? [1]

[1] http://docs.ceph.com/docs/master/rados/api/python/#rados.Rados.version

File: gwcli/ceph.py [hunk 2]
```diff
@@ -256,10 +301,6 @@ def populate(self):
             name = pool['pool_name']
             pools[name] = pool
 
-        # # Get the pools defined
```
Comment (dillaman) [inline gwcli/ceph.py]: While you are changing this -- perhaps hide cache tiers?

File: rbd-target-api.py [hunk 3]
```diff
@@ -1368,19 +1408,27 @@ def pre_reqs_errors():
...
+
     required_rpms = [
-        # {"name": "device-mapper-multipath",
-        #  "version": "0.4.9",
-        #  "release": "99.el7"},
         {"name": "python-rtslib",
-         "version": "2.1.fb57",
-         "release": "5.el7"}
+         "version": "2.1.fb64",
+         "release": "0.1.20170815.gitec364f3.el7cp"},
```
Comment (dillaman) [inline rbd-target-api.py]: The ```el7cp``` suffixes are downstream specific -- does the version checking still work if you just leave off the suffix?

File: gwcli/ceph.py (hunk 1 above)
Comment (pcuzner) [reply gwcli/ceph.py]: running ceph with -c to point to the appropriate cluster I get that clusters version. Maybe I'm doing something wrong, but this is what I get with the rados binding
>>> import rados
>>> r=rados.Rados(conffile='/etc/ceph/ceph.conf')
>>> str(r.version())
'0.69.1'

whereas this is what I'm after

e.g. 
/disks> /clusters/ceph/ info 
Cluster name: ceph
Ceph version: 12.1.3
Health      : HEALTH_OK

MONs :    3 (quorum ceph-1,ceph-2,ceph-3)
OSDs :    4 (4 up, 4 in)
Pools:    3
Raw capacity: 85G

Config : /etc/ceph/ceph.conf
Keyring: /etc/ceph/ceph.client.admin.keyring

/disks> exit
[root@rh7-gw1 ~]# ceph --version
ceph version 12.1.4 (a5f84b37668fc8e03165aaf5cbb380c78e4deba4) luminous (rc)

the gateway is on 12.1.4, but the cluster is running 12.1.3

Comment (dillaman) [reply gwcli/ceph.py]: Hmm -- turns out that method just returns the constants defined in librados.h, which seems pointless. Disregard.

File: gwcli/ceph.py (hunk 2 above)
Comment (pcuzner) [reply gwcli/ceph.py]: I can add this to my to-do sure - but for this PR I just want to get the health issues fixed

File: rbd-target-api.py (hunk 3 above)
Comment (pcuzner) [reply rbd-target-api.py]: looks like it. as long as the installed version meets or exceeds the version given it should be fine. will update

File: gwcli/ceph.py [hunk 4]
```diff
@@ -171,38 +149,99 @@ def ui_command_refresh(self):
...
 
-        self.ceph_status = json.loads(buf_s)
-        self.health_status = self.ceph_status['health']['overall_status']
+        status = json.loads(buf_s)
+        self.ceph_status = status
+
+        # assume a luminous or above ceph cluster
+        self.health_status = status['health'].get('status',
+                                                  status['health'].get('overall_status'))
+        self.health_list = [check.get('summary') for check in
```
Comment (dillaman) [inline gwcli/ceph.py]: This fails in my test environment w/ v12.1: the contents of ```check.get('summary')``` is ```{u'message': u'application not enabled on 1 pool(s)'}```, which results in a ```TypeError: 'NoneType' object is not iterable``` exception

Comment (pcuzner) [reply gwcli/ceph.py]: I disabled rbd from my pool, and see the same message from ceph but don't get the exception.

Is mon_health_preluminous_compat=true set ?

**Pull request merged**